packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart

```diff
@@ -123,43 +123,44 @@ class User extends UserInfo<auth_interop.UserJsImpl> {
   /// It forces refresh regardless of token expiration if [forceRefresh]
   /// parameter is `true`.
   Future<String> getIdToken([bool forceRefresh = false]) =>
-      handleThenable(jsObject.getIdToken(forceRefresh));
+      jsObject.getIdToken(forceRefresh.toJS).toDart as Future<String>;
 
   /// Links the user account with the given credentials, and returns any
   /// available additional user information, such as user name.
   Future<UserCredential> linkWithCredential(
           auth_interop.OAuthCredential? credential) =>
-      handleThenable(auth_interop.linkWithCredential(jsObject, credential))
+      (auth_interop.linkWithCredential(jsObject, credential).toDart
+              as Future<auth_interop.UserCredentialJsImpl>)
           .then(UserCredential.fromJsObject);
 
   /// Links the user account with the given [phoneNumber] in E.164 format
   /// (e.g. +16505550101) and [applicationVerifier].
   Future<ConfirmationResult> linkWithPhoneNumber(
           String phoneNumber, ApplicationVerifier applicationVerifier) =>
-      handleThenable(
-        auth_interop.linkWithPhoneNumber(
-            jsObject, phoneNumber, applicationVerifier.jsObject),
-      ).then(ConfirmationResult.fromJsObject);
+      (auth_interop
+              .linkWithPhoneNumber(
+                  jsObject, phoneNumber.toJS, applicationVerifier.jsObject)
+              .toDart as Future<auth_interop.ConfirmationResultJsImpl>)
+          .then(ConfirmationResult.fromJsObject);
 
   /// Links the authenticated [provider] to the user account using
   /// a pop-up based OAuth flow.
   /// It returns the [UserCredential] information if linking is successful.
   Future<UserCredential> linkWithPopup(AuthProvider provider) =>
-      handleThenable(auth_interop.linkWithPopup(jsObject, provider.jsObject))
-          .then(UserCredential.fromJsObject);
+      auth_interop.linkWithPopup(jsObject, provider.jsObject).toDart
+          as Future<UserCredential>;
 
   /// Links the authenticated [provider] to the user account using
   /// a full-page redirect flow.
-  Future<void> linkWithRedirect(AuthProvider provider) => handleThenable(
-      auth_interop.linkWithRedirect(jsObject, provider.jsObject));
+  Future<void> linkWithRedirect(AuthProvider provider) =>
+      auth_interop.linkWithRedirect(jsObject, provider.jsObject).toDart;
```

Comment:
```suggestion
  Future<void> linkWithRedirect(AuthProvider provider) =>
      auth_interop.linkWithRedirect(jsObject, provider.jsObject).toDart as Future<void>;
```
Is this correct?

Reply:
When you run .toDart, it is interpreted as Future<JSAny> which includes Future<void> so casting isn't necessary